packages/flutter_tools/lib/src/hot.dart

```diff
@@ -274,30 +321,34 @@ class HotRunner extends ResidentRunner {
 
     printStatus('Application running.');
 
-    setupTerminal();
-
     registerSignalHandlers();
 
     printStatus('Finishing file synchronization...');
     // Finish the file sync now.
     await _updateDevFS();
 
+    setupTerminal();
+
     return waitForAppToFinish();
   }
 
   @override
-  Future<Null> handleTerminalCommand(String code) async {
-    final String lower = code.toLowerCase();
-    if ((lower == 'r') || (code == AnsiTerminal.KEY_F5)) {
-      // F5, restart
-      if ((code == 'r') || (code == AnsiTerminal.KEY_F5)) {
-        // lower-case 'r'
-        await _reloadSources();
-      } else {
-        // upper-case 'R'.
-        await _restartFromSources();
-      }
-    }
+  List<Command> buildCommandList() {
+    List<Command> cmds = <Command>[];
+    cmds.add(new HelpCommand(this));
+    cmds.add(new PrintCommand(this));
+    cmds.add(new QuitCommand(this));
+    cmds.add(new ReloadCommand(this)..alias = 'r');
```

Comment:
if i understand this correctly, this changes the command for reloading from "r" to "r enter". I think this is a mistake. Hot reloading is by many orders of magnitude the most common command and triggering it should not require more than one key press, and ideally not a function key.